src/sqlfluff/utils/reflow/reindent.py

```diff
@@ -1546,12 +1546,108 @@ def _match_indents(
     return matched_indents
 
 
+def _fix_long_line_with_comment(
+    line_buffer: ReflowSequenceType,
+    elements: ReflowSequenceType,
+    current_indent: str,
+    line_length_limit: int,
+    last_indent_idx: Optional[int],
+    trailing_comments: str = "before",
+) -> Tuple[ReflowSequenceType, List[LintFix]]:
+    """Fix long line by moving trailing comments if possible."""
+    # If the comment contains a noqa, don't fix it. It's unsafe.
+    if "noqa" in line_buffer[-1].segments[-1].raw:
+        reflow_logger.debug("    Unfixable because noqa unsafe to move.")
+        return elements, []
+
+    # If the comment is longer than the limit _anyway_, don't move
+    # it. It will still be too long.
+    if len(line_buffer[-1].segments[-1].raw) + len(current_indent) > line_length_limit:
+        reflow_logger.debug("    Unfixable because comment too long anyway.")
+        return elements, []
+
+    comment_seg = line_buffer[-1].segments[-1]
+    first_seg = line_buffer[0].segments[0]
+    last_elem_idx = elements.index(line_buffer[-1])
+
+    assert trailing_comments in (
+        "after",
+        "before",
+    ), f"Unexpected value for `trailing_comments`: {trailing_comments!r}"
+
+    # The simpler case if if we're moving the comment to the line
+    # _after_. In that case we just coerce the point before it to
+    # be an indent.
+    if trailing_comments == "after":
+        anchor_point = cast(ReflowPoint, line_buffer[-2])
+        results, new_point = anchor_point.indent_to(current_indent, before=comment_seg)
+        # Mutate elements
+        elements[last_elem_idx - 1] = new_point
```

Comment:
Similar to the other PR, we're mutating **and** returning a list. Could we mutate without returning?

Reply:
In this case, we actually _need_ to return it (see the later clause of the function), but I can update this section so that is doesn't mutate, and then we're entirely in _just return the value_.